File: cookblog-frontend/src/app/api/resources/ingredient.resource.ts
import { QuantityResource } from './quantity.resource';

export class IngredientResource {
  private readonly _id: number | null;
  private readonly _name: string;
  private readonly _quantity: QuantityResource;

  constructor(id: number, name: string, quantity: QuantityResource) {
    this._id = id;
    this._name = name;
    this._quantity = quantity;
  }

  get id() {
    return this._id;
  }

  get name() {
    return this._name;
  }

  get quantity() {
    return this._quantity;
  }
}

File: cookblog-backend/src/main/java/com/example/cookblog/dto/resources/IngredientResource.java
package com.example.cookblog.dto.resources;

import com.fasterxml.jackson.annotation.JsonAutoDetect;
import jakarta.validation.Valid;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Null;
import jakarta.validation.constraints.Size;
import lombok.Builder;
import lombok.extern.jackson.Jacksonized;

@Builder
@Jacksonized
@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
public record IngredientResource(
        @Null Long id,
        @NotBlank @Size(max = 255) String name,
        @NotNull @Valid QuantityResource quantity
) {

}